### PR TITLE
PROXY-7: Add logic to your app

### PR DESCRIPTION
**Pull request by DmitryPopov1988**

### CODE REVIEW

File: src/client/nasa_client.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        response.on('data', chunk => {
+            data += chunk;
+        });
+
+        response.on('end', () => {
+            try {
+
+                const jsonData = JSON.parse(data);
+                res.set("Content-Type", "application/json");
```
Comment (plutsenka) [inline src/client/nasa_client.js]: You can use `app.use(express.json());` in your app.js file, so header will be always set by library

File: src/client/nasa_client.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const buildUrl = (startDate, endDate) => {
+    const baseUrl = config.nasaApiConfig.url;
+    const apiKey = config.nasaApiConfig.key;
+    return `${baseUrl}?start_date=${startDate}&end_date=${endDate}&api_key=${apiKey}`;
+};
+
+
+export const client = (req, res, startDate, endDate) => {
+
```
Comment (plutsenka) [inline src/client/nasa_client.js]: extra empty line

File: src/app.js [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+
+    if (!startDate) {
+        startDate = new Date().toISOString().split('T')[0];
+    }
+
+    if (!endDate) {
+        endDate = new Date().toISOString().split('T')[0];
+    }
+
+    client(req, res, startDate, endDate);
```
Comment (plutsenka) [inline src/app.js]: Name is not very informative. It's ok for right now, but it can become a trouble later

File: src/app.js [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
 
 export const app = express();
 
-app.get('/meteors', client);
+app.get('/meteors', (req, res) => {
+    let startDate = req.query.start_date;
+    let endDate = req.query.end_date;
+
+    if (!startDate) {
+        startDate = new Date().toISOString().split('T')[0];
```
Comment (plutsenka) [inline src/app.js]: You can use libraries (e.g date-fns) to perform such kind of manipulations

File: src/client/nasa_client.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const request = https.get(buildUrl(startDate, endDate), response => {
```
Comment (plutsenka) [inline src/client/nasa_client.js]: Such kind of work can be done with e.g. axios library

File: src/service/mapper.js
Comment (plutsenka) [inline src/service/mapper.js:1]: I'd suggest use more specific names - later more mappers can be needed and informative naming will help

File: src/client/nasa_client.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import {config} from "../config/config.js";
+import {mapAsteroids} from '../service/mapper.js';
+
+const buildUrl = (startDate, endDate) => {
+    const baseUrl = config.nasaApiConfig.url;
+    const apiKey = config.nasaApiConfig.key;
+    return `${baseUrl}?start_date=${startDate}&end_date=${endDate}&api_key=${apiKey}`;
+};
+
+
```
Comment (plutsenka) [inline src/client/nasa_client.js]: Extra empty line

File: src/client/nasa_client.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    const request = https.get(buildUrl(startDate, endDate), response => {
+        let data = '';
+
+        response.on('data', chunk => {
+            data += chunk;
+        });
+
+        response.on('end', () => {
+            try {
+
```
Comment (plutsenka) [inline src/client/nasa_client.js]: Extra empty line

File: src/service/mapper.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    for (const date in nasaAsteroids.near_earth_objects) {
+        const asteroids = nasaAsteroids.near_earth_objects[date];
+        const mappedAsteroidsForDate = asteroids.map(mapAsteroid);
+        ans.push({date: date, asteroids: mappedAsteroidsForDate});
+    }
+
+    return ans;
+};
+
+
```
Comment (plutsenka) [inline src/service/mapper.js]: Extra empty line

File: src/app.js [hunk 9]
```diff
@@ -1,11 +1,24 @@
...
-import {client} from "./client/meteor_nasa_client.js";
+import {config} from "./config/config.js";
+import {client} from "./client/nasa_client.js";
 
 
 export const app = express();
 
-app.get('/meteors', client);
+app.get('/meteors', (req, res) => {
+    let startDate = req.query.start_date;
```
Comment (plutsenka) [inline src/app.js]: Such kind of operations are better to be moved e.g. to controller

File: src/client/nasa_client.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+                const jsonData = JSON.parse(data);
+                res.set("Content-Type", "application/json");
+                res.send(mapAsteroids(jsonData));
+            } catch (error) {
+                res.status(500).json({error: 'Error parsing JSON', message: error.message});
```
Comment (mariaYurcheuskaya) [inline src/client/nasa_client.js]: you can use middleware approach for dealing with errors(you can find example in our chat)

**Review by plutsenka — APPROVED** [review]
Well done!

**Comment by DmitryPopov1988** [comment]
> Well done!

**Pull request merged**